### PR TITLE
MM-48824 - move onboarding tasklist to logged in component

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR moves the OnboardingTaskList component from the root component to the LoggedIn component to make sure that all the requests made from this component are only made once the user has logged in which is where actually makes sense.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48824

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/root/root.tsx [hunk 1]
```diff
@@ -593,9 +592,6 @@ export default class Root extends React.PureComponent<Props, State> {
                                 />
                                 <RootRedirect/>
                             </Switch>
-                            <CompassThemeProvider theme={this.props.theme}>
-                                <OnBoardingTaskList/>
-                            </CompassThemeProvider>
                         </>
```
Comment (M-ZubairAhmed) [inline components/root/root.tsx:595]: nit : we dont need the fragments here anymore

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by neallred — APPROVED** [review]
LGTM, thanks @pvev !

File: components/root/root.tsx (hunk 1 above)
Comment (pvev) [reply components/root/root.tsx:595]: nice catch, thanks!

File: components/logged_in/logged_in.tsx [hunk 2]
```diff
@@ -152,7 +156,14 @@ export default class LoggedIn extends React.PureComponent<Props> {
             }
         }
 
-        return this.props.children;
+        return (
+            <>
+                <CompassThemeProvider theme={this.props.theme}>
+                    <OnBoardingTaskList/>
+                </CompassThemeProvider>
```
[the remark this answers is not in the document]
Comment (pvev) [reply components/logged_in/logged_in.tsx]: good point, it was not necessary for the onboarding to appear in the places handled by the LoggedInHftRoute so I moved it to the LoggedInRoute. Thanks for the observations @M-ZubairAhmed 💯

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]

File: components/logged_in/logged_in.tsx [hunk 3]
```diff
@@ -32,6 +33,7 @@ export type Props = {
     children?: React.ReactNode;
     mfaRequired: boolean;
     enableTimezone: boolean;
+    theme: Theme;
```
Comment (pvev) [inline components/logged_in/logged_in.tsx:36]: this theme is no longer needed because the OnboardingTaskList was moved to the LoggedInRoute Component. Will send a new PR https://github.com/mattermost/mattermost-webapp/pull/11781 to remove this unnecessary change. Same goes for the Fragment on this same file on line 158.
